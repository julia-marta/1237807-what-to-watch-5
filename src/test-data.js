import React from 'react';
import PropTypes from "prop-types";
export const noop = () => {};

export const promoFilm = {
  id: 1,
  name: `The Grand Budapest Hotel`,
  posterImage: `the-grand-budapest-hotel-poster.jpg`,
  backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#000000`,
  genre: `Drama`,
  released: 2014,
  isFavorite: false,
};

export const filmCard = {
  id: 1,
  name: `The Grand Budapest Hotel`,
  genre: `Drama`,
  previewImage: `the-grand-budapest-hotel.jpg`,
  previewVideoLink: `https://trailers.apple.com/movies/universal/krampus/krampus-tlr1_h480p.mov`,
};

export const film = {
  id: 1,
  name: `The Grand Budapest Hotel`,
  posterImage: `the-grand-budapest-hotel-poster.jpg`,
  backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#000000`,
  genre: `Drama`,
  released: 2014,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  rating: 8.9,
  scoresCount: 240,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  runTime: 99,
  isFavorite: true,
};

export const reviews = [
  {
    id: 1,
    user: {
      id: 4,
      name: `Kate Muir`,
    },
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`,
  },
  {
    id: 2,
    user: {
      id: 5,
      name: `Matthew Lickona`,
    },
    rating: 7.2,
    comment: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    date: `2019-05-18T14:13:56.569Z`,
  }
];

export const genres = [`All genres`, `Drama`, `Music`, `Biography`, `Survival`, `Thriller`, `Crime`];

export const defaultState = {
  DATA: {
    films: [filmCard, filmCard, filmCard, filmCard, filmCard, filmCard, filmCard],
    promo: promoFilm,
    currentFilm: film,
    currentReviews: reviews,
  },
  FILTER: {
    genre: `Drama`,
    cardsCount: 8,
  },
  USER: {
    status: `NOT_AUTHORIZED`,
    name: `userName`,
    avatar: `avatar.jpg`,
    reviewStatus: `NOT_SAVING`,
  },
};

export const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

export const SimpleMockComponent = () => <div />;
