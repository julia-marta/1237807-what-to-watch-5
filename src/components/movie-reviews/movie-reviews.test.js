import React from 'react';
import renderer from 'react-test-renderer';
import MovieReviews from "./movie-reviews";
import {reviews} from "../../test-data";

it(`should MovieReviews tab render correctly`, () => {
  const tree = renderer
    .create(
        <MovieReviews reviews={reviews} />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
