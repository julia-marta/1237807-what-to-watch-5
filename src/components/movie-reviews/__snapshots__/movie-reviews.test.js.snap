// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should MovieReviews tab render correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Kate Muir
          </cite>
          <time
            className="review__date"
            dateTime="2019-05-08T14:13:56.569Z"
          >
            May 08, 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        8.9
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Matthew Lickona
          </cite>
          <time
            className="review__date"
            dateTime="2019-05-18T14:13:56.569Z"
          >
            May 18, 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        7.2
      </div>
    </div>
  </div>
</div>
`;
